database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -154,6 +154,58 @@ public <T extends BaseEntity> T addNewRecordToTable(T entity) {
         return entity;
     }
 
+    @Override
+    public <T extends BaseEntity> Iterable<T> addNewRecordsToTable(Class<? extends BaseEntity> entityClass,
+                                                                   List<T> entities) {
+        for (T t : entities) {
+            if (t.getId() != null) {
+                throw new IdProvidedManuallyException(ID_PROVIDED_MANUALLY);
+            }
+        }
+
+        String tableName = entityClass.getSimpleName();
+
+        Connection connection = connectionPool.getConnection();
+
+        try {
+            if (checkTableExists(databaseName, tableName, connection)) {
+                String insertSQL = generateInsertMultipleLinesSQL(entities, tableName);
+                try (PreparedStatement preparedStatement = connection.prepareStatement(insertSQL,
+                        Statement.RETURN_GENERATED_KEYS)) {
+
+                    setPreparedStatementValuesForMultipleInsert(preparedStatement, entities);
+
+                    LOG.info("Executing SQL: {}", insertSQL);
+
+                    int affectedRows = preparedStatement.executeUpdate();
```

Comment:
Please use `executeBatch(`) to insert all records in one operation. Check how to add records in batches, you can't just execute query with multiple insert operators.